BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/External/ExceptionDetailsImplementation.cs

```diff
@@ -21,7 +22,7 @@ internal static ExceptionDetails CreateWithoutStackInfo(Exception exception, Exc
             {
                 id = exception.GetHashCode(),
                 typeName = exception.GetType().FullName,
-                message = exception.Message,
+                message = Utils.PopulateRequiredStringValue(exception.Message, "message", typeof(ExceptionTelemetry).FullName),
```

Comment:
nit: note that this helper method is using `string.IsNullOrEmpty`. 
https://github.com/microsoft/ApplicationInsights-dotnet/blob/bd8b0c25712e6f3c5275ae06b0f0e9331340fee4/BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Utils.cs#L44-L49

I see in your test cases you passed in a string with a space `" "`. Note that this helper method won't catch this.

Reply:
I fixed the test to use the parameter and it failed for the null and whitespace examples. 

- The null failure was due to the `Message` property on `Exception` returning a default if the `_message` field is null. I've added a custom exception for the test that overrides Message and returns the provided message as is.
- The empty string failure was down to the `.IsNullOrEmpty` so I've added a new, similar helper method that instead uses the provided `IsNullOrWhitespace` extension. In cases where it is null/whitespace I've used the same logging method, but can add another slightly different one with a different EventId if you prefer?